lib/appsignal/transaction.rb

```diff
@@ -126,11 +127,86 @@ def set_tags(given_tags = {})
       @tags.merge!(given_tags)
     end
 
+    # Set a custom action name for this transaction.
+    #
+    # When using an integration such as the Rails or Sinatra AppSignal will try
+    # to find the action name from the controller or endpoint for you.
+    #
+    # If you want to customize the action name as it appears on AppSignal.com
+    # you can use this method. This overrides the action name AppSignal
+    # generates in an integration.
+    #
+    # @example in a Rails controller
+    #   class SomeController < ApplicationController
+    #     before_action :set_appsignal_action
+    #
+    #     def set_appsignal_action
+    #       Appsignal.set_action("DynamicController#dynamic_method")
+    #     end
+    #   end
+    #
+    # @param action [String] the action name to set.
+    # @return [void]
     def set_action(action)
       return unless action
+      @action = action
       @ext.set_action(action)
     end
 
+    # Set an action name only if there is no current action set.
+    #
+    # Commonly used by AppSignal integrations so that they don't override
+    # custom action names.
+    #
+    # @example
+    #   Appsignal.set_action("foo")
+    #   Appsignal.set_action_if_nil("bar")
+    #   # Transaction action will be "foo"
+    #
+    # @param action [String]
+    # @return [void]
+    # @see #set_action
+    def set_action_if_nil(action)
```

Comment:
@thijsc do we want to document this method as private/internal use or do we want to allow people to use this method in their own apps?